x-pack/plugins/lens/public/indexpattern_plugin/indexpattern.tsx

```diff
@@ -125,45 +129,73 @@ export function IndexPatternDimensionPanel(props: IndexPatternDimensionPanelProp
 
   const selectedColumn: IndexPatternColumn | null = props.state.columns[props.columnId] || null;
 
+  function canHandleDrop() {
```

Comment:
Nit: the typescript-y way for this situation is a custom typeguard, something like this:

```ts
  function isIndexPatternField(dragging: unknown) : dragging is IndexPatternField {
    return typeof dragging === "object" && dragging !== null && 'type' in dragging;
  }

  function canHandleDrop() {
    const { dragging } = props.dragDropContext;

    return (
      isIndexPatternField(dragging) &&
      filteredColumns.some(({ sourceField }) => sourceField === dragging.name)
    );
  }
```